### ISSUE TITLE
Maps and init function not playing nice: Redux

### ISSUE THREAD

**Issue by ribasushi** [tinygo-org/tinygo#1307]
**TLDR**
The problem described in https://github.com/tinygo-org/tinygo/issues/545 appears to have not been fully fixed.

**Version**
`tinygo version 0.14.0 darwin/amd64 (using go version go1.14.7 and LLVM version 10.0.0)`

**Failure**
```
~$ tinygo run tiny.go 
panic: interp: load from a bitcast

goroutine 1 [running]:
github.com/tinygo-org/tinygo/interp.(*LocalValue).Load(0xc0057c6480, 0xc0057c6480)
        /Users/admin/devel/tinygo/interp/values.go:61 +0x220
github.com/tinygo-org/tinygo/interp.(*LocalValue).Load(0xc0057c6470, 0x1d)
        /Users/admin/devel/tinygo/interp/values.go:58 +0x12f
github.com/tinygo-org/tinygo/interp.(*frame).evalBasicBlock(0xc003605730, 0x37fb72a0, 0x37fb6f50, 0xc0057b65b8, 0x8, 0x0, 0x0, 0xc0057ac860, 0x1, 0x1, ...)
        /Users/admin/devel/tinygo/interp/frame.go:98 +0xd684
github.com/tinygo-org/tinygo/interp.(*evalPackage).function(0xc0036074c0, 0xb422508, 0xc00467d200, 0x6, 0x8, 0xc0057b65b8, 0x8, 0x8, 0xc00573deb0, 0x0)
        /Users/admin/devel/tinygo/interp/interp.go:113 +0x238
github.com/tinygo-org/tinygo/interp.(*frame).evalBasicBlock(0xc0036064f0, 0x388f8750, 0x388f8640, 0x44be553, 0x4, 0x0, 0x0, 0xc0057ac750, 0x1, 0x1, ...)
        /Users/admin/devel/tinygo/interp/frame.go:570 +0x6463
github.com/tinygo-org/tinygo/interp.(*evalPackage).function(0xc0036074c0, 0xb41ea98, 0xc0021c8020, 0x2, 0x2, 0x44be553, 0x4, 0x2, 0xc0057b2fb0, 0x0)
        /Users/admin/devel/tinygo/interp/interp.go:113 +0x238
github.com/tinygo-org/tinygo/interp.(*frame).evalBasicBlock(0xc0036072b0, 0x388f7380, 0x388f6170, 0x0, 0x0, 0x0, 0x0, 0xc0057ac678, 0x1, 0x1, ...)
        /Users/admin/devel/tinygo/interp/frame.go:570 +0x6463
github.com/tinygo-org/tinygo/interp.(*evalPackage).function(0xc0036074c0, 0xb41e9b8, 0xc0036074d8, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/admin/devel/tinygo/interp/interp.go:113 +0x238
github.com/tinygo-org/tinygo/interp.Run(0xb2113a0, 0x0, 0x0, 0x0)
        /Users/admin/devel/tinygo/interp/interp.go:85 +0x7fe
github.com/tinygo-org/tinygo/builder.Build(0x7ffeefbff93b, 0x7, 0x44be5ff, 0x4, 0xc000270c00, 0xc003607c80, 0x0, 0x0)
        /Users/admin/devel/tinygo/builder/build.go:51 +0x20d
main.Run(0x7ffeefbff93b, 0x7, 0xc00020a0e0, 0xe, 0x0)
        /Users/admin/devel/tinygo/main.go:442 +0xaf
main.main()
        /Users/admin/devel/tinygo/main.go:944 +0x105a```
```

**Expectation**
```
~$ go run tiny.go 
name1341341341
```

**Testcase**
```
package main

import "fmt"

type thingDef struct {
	label string
	thing []byte
}

var veryLongListOfAvailableThings = map[string]thingDef{
	"name1":          {},
	"name2":          {},
	"name1341341341": {},
}

func init() {
	for k, v := range veryLongListOfAvailableThings {
		v.label = k
		veryLongListOfAvailableThings[k] = v
	}
}

func main() {
	fmt.Println(veryLongListOfAvailableThings["name1341341341"].label)
}
```

**Comment by aykevl** [comment]
Hopefully #1430 will fix this.

**Comment by wbourne0** [comment]
I'm also having this issue (also mentioned this in an issue that I created before realizing that there was already an issue for the same thing): 

Defining a map outside of a function causes a compiler error.

This is with with an arduino mega 2560.  It should be llvm11. 

<details>
   <summary>Stack trace</summary>

```go
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x7f95325d885f pc=0x7f942f707dde]

runtime stack:
runtime.throw(0x7f969d, 0x2a)
	/usr/lib/go/src/runtime/panic.go:1116 +0x72
runtime.sigpanic()
	/usr/lib/go/src/runtime/signal_unix.go:726 +0x4ac

goroutine 1 [syscall]:
runtime.cgocall(0x74a610, 0xc00183c518, 0xc000026000)
	/usr/lib/go/src/runtime/cgocall.go:133 +0x5b fp=0xc00183c4e8 sp=0xc00183c4b0 pc=0x421e3b
tinygo.org/x/go-llvm._Cfunc_LLVMGetConstOpcode(0x7f93c4754d70, 0xc000000000)
	_cgo_gotypes.go:5389 +0x49 fp=0xc00183c518 sp=0xc00183c4e8 pc=0x664369
tinygo.org/x/go-llvm.Value.Opcode.func1(0x7f93c4754d70, 0xc00183c578)
	/home/allawesome497/go/pkg/mod/tinygo.org/x/go-llvm@v0.0.0-20201104183921-570e7a6841d9/ir.go:904 +0x53 fp=0xc00183c548 sp=0xc00183c518 pc=0x67f3d3
tinygo.org/x/go-llvm.Value.Opcode(0x7f93c4754d70, 0x67e8f3)
	/home/allawesome497/go/pkg/mod/tinygo.org/x/go-llvm@v0.0.0-20201104183921-570e7a6841d9/ir.go:904 +0x2b fp=0xc00183c568 sp=0xc00183c548 pc=0x670b8b
github.com/tinygo-org/tinygo/interp.(*MapValue).PutBinary(0xc001d30c40, 0xc0026089a0, 0xc0026089c0, 0xc002616968, 0x1)
	/home/allawesome497/Builds/tinygo/interp/values.go:372 +0x5b fp=0xc00183c600 sp=0xc00183c568 pc=0x7015fb
github.com/tinygo-org/tinygo/interp.(*frame).evalBasicBlock(0xc00183d0d0, 0x7f93c46404d0, 0x7f93c4640000, 0x0, 0x0, 0x0, 0x0, 0xc002618470, 0x1, 0x1, ...)
	/home/allawesome497/Builds/tinygo/interp/frame.go:342 +0x4b65 fp=0xc00183d040 sp=0xc00183c600 pc=0x6f5105
github.com/tinygo-org/tinygo/interp.(*evalPackage).function(0xc00183d2d8, 0x7f93f4017e98, 0xc00183d2f0, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/allawesome497/Builds/tinygo/interp/interp.go:113 +0x249 fp=0xc00183d1f8 sp=0xc00183d040 pc=0x6fd289
github.com/tinygo-org/tinygo/interp.Run(0x1943060, 0x0, 0x0, 0x0)
	/home/allawesome497/Builds/tinygo/interp/interp.go:85 +0x7a8 fp=0xc00183d460 sp=0xc00183d1f8 pc=0x6fcd28
github.com/tinygo-org/tinygo/builder.Build(0x7ffc720eb11d, 0x1, 0x7e6640, 0x4, 0xc000240570, 0xc00183db78, 0x0, 0x0)
	/home/allawesome497/Builds/tinygo/builder/build.go:65 +0x21a fp=0xc00183db10 sp=0xc00183d460 pc=0x72b69a
main.Flash(0x7ffc720eb11d, 0x1, 0x7ffc720eb110, 0xc, 0xc000238000, 0x0, 0x1e)
	/home/allawesome497/Builds/tinygo/main.go:234 +0x1a5 fp=0xc00183dbc0 sp=0xc00183db10 pc=0x73d3c5
main.main()
	/home/allawesome497/Builds/tinygo/main.go:954 +0x10c5 fp=0xc00183df88 sp=0xc00183dbc0 pc=0x740b45
runtime.main()
	/usr/lib/go/src/runtime/proc.go:204 +0x209 fp=0xc00183dfe0 sp=0xc00183df88 pc=0x4561a9
runtime.goexit()
	/usr/lib/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc00183dfe8 sp=0xc00183dfe0 pc=0x487bc1
make: *** [Makefile:11: main] Error 2

```

</details>
<details>
    <summary>Source code</summary>
<details>
    <summary>main.go:</summary>

```go
package main

import (
	"machine"
)

var (
	output = machine.PinConfig{
		Mode: machine.PinOutput,
	}
	input = machine.PinConfig{
		Mode: machine.PinInput,
	}
	tubeTab = []byte{
		0x3f, 0x06, 0x5b, 0x4f, 0x66, 0x6d, 0x7d, 0x07, 0x7f, 0x6f, //0-9
		0x77, 0x7c, 0x39, 0x5e, 0x79, 0x71, //'A', 'B', 'C', 'D', 'E', 'F',
		0xbf, 0x86, 0xdb, 0xcf, 0xe6, 0xed, 0xfd, 0x87, 0xff, 0xef, //0.-9.
		0xf7, 0xfc, 0xb9, 0xde, 0xf9, 0xf1, //'A.', 'B.', 'C.', 'D.', 'E.', 'F.',
		0, 0x40, //' ','-'
	}

	numMap = map[byte]byte{
		'1': 0x06,
		'2': 0x5b,
		'3': 0x4f,
		'4': 0x66,
		'5': 0x6d,
		'6': 0x7d,
		'7': 0x07,
		'8': 0x7f,
		'9': 0x6f,
		'0': 0x3f,
		' ': 0,
		'_': 0x8,
		'-': 0x40,
		'F': 0x71,
		'f': 0x71,
		'e': 0x79,
		'E': 0x79,
		'd': 0x5e,
		'C': 0x39,
		'c': 0x58,
		'b': 0x7c,
		'A': 0x77,
		'o': 0x5c,
		'O': 0x3f,
	}
)

func main() {

	clkPin.Configure(output)
	dataPin.Configure(output)
	clkPin.Low()
	dataPin.Low()

	for {
		in := []byte{}
		for {
			for machine.UART0.Buffered() == 0 {
			}

			v, err := machine.UART0.ReadByte()

			if err != nil {
				println("err:", err.Error())
				continue
			}

			if v == '\b' {
				machine.UART0.Write([]byte{'\b', 127})
				if len(in) > 0 {
					in = in[:len(in)-1]
				}
				continue
			}

			if err = machine.UART0.WriteByte(v); err != nil {
				println("err:", err.Error())
				continue
			}

			if v == '\r' {
				machine.UART0.WriteByte('\n')
				break
			}

			in = append(in, v)
		}

		dispIndex := 0
		valid := true

		disp := [4]byte{}

		for i := 0; i < len(in); i++ {

			if dispIndex == len(disp) {
				break
			}

			char := in[i]

			if char == '.' {
				if dispIndex == 0 {
					disp[dispIndex] = 0x80
				} else {
					disp[dispIndex-1] |= 0x80
				}
				dispIndex++
				continue
			}

			if v, ok := numMap[char]; ok {
				disp[dispIndex] = v
				dispIndex++
			} else {
				println("invalid value:", string(char), char)
				valid = false
				break
			}
		}

		if !valid {
			continue
		}

		write(disp)
	}

}
```

</details>

<details>
    <summary>segdisp.go:</summary>

```go
package main

import (
	"machine"
)

const (
	brightness0 = iota
	brightness1
	brightness2
	brightness3
	brightness4
	brightness5
	brightness6
	brightness7
)

const (
	clkPin    = machine.D26
	dataPin   = machine.D28
	sensorPin = machine.D24

	// TM1637
	addrAuto  = 0x40
	addrFixed = 0x44
	startAddr = 0xc0

	segdisOn  = 0x88
	segdisOff = 0x80
)

func start() {
	clkPin.High()
	dataPin.High()
	dataPin.Low()
	clkPin.Low()
}

func stop() {
	clkPin.Low()
	dataPin.Low()
	clkPin.High()
	dataPin.High()
}

func writeByte(v byte) {
	for i := 0; i < 8; i++ {
		clkPin.Low()
		dataPin.Set((v>>i)&0x01 == 0x01)
		clkPin.High()
	}

	clkPin.Low()
	dataPin.High()
	clkPin.High()
	dataPin.Configure(input)

	for cnt := 0; dataPin.Get(); cnt++ {
		if cnt+1 == 200 {
			dataPin.Configure(output)
			dataPin.Low()
			cnt = 0
		}
	}

	dataPin.Configure(output)
}

func write(v [4]byte) {
	start()
	writeByte(addrAuto)
	stop()
	start()
	writeByte(startAddr)

	for i := 0; i < 4; i++ {
		writeByte(v[i])
	}

	println()

	stop()
	start()
	writeByte(segdisOn + brightness7)
	stop()
}
```

</details>
</details>

**Comment by deadprogram** [comment]
Is this still an issue with the latest TinyGo since #1430 was previously merged?

**Comment by niaow** [comment]
This is now fixed, possibly by #1430.
